packages/ios_platform_images/ios/Classes/IosPlatformImagesPlugin.m

```diff
@@ -40,6 +40,115 @@ + (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar> *)registrar {
+
+        // Up to 3 rgb values for primary, seconday and tertiary colors.
+        // see
+        // https://developer.apple.com/documentation/uikit/uiimagesymbolconfiguration/3810054-configurationwithpalettecolors
+        NSArray *rgbaValuesList = call.arguments[4];
+
+        NSMutableArray *colorArray = [[NSMutableArray alloc] init];
+
+        for (int i = 0; i < [rgbaValuesList count]; i += 4) {
+          UIColor *primaryColor = [UIColor colorWithRed:[rgbaValuesList[i] doubleValue]
+                                                  green:[rgbaValuesList[i + 1] doubleValue]
+                                                   blue:[rgbaValuesList[i + 2] doubleValue]
+                                                  alpha:[rgbaValuesList[i + 3] doubleValue]];
+          [colorArray addObject:primaryColor];
+        }
+
+        UIImageSymbolScale scale = UIImageSymbolScaleDefault;
+
+        switch ([scaleIndex integerValue]) {
+          case 0:
+            scale = UIImageSymbolScaleSmall;
+            break;
+          case 1:
+            scale = UIImageSymbolScaleMedium;
+            break;
+          case 2:
+            scale = UIImageSymbolScaleLarge;
+            break;
+          default:
+            scale = UIImageSymbolScaleDefault;
+            break;
+        }
+
+        UIImageSymbolWeight weight = UIImageSymbolWeightRegular;
+
+        switch ([weightIndex integerValue]) {
+          case 0:
+            weight = UIImageSymbolWeightUltraLight;
+            break;
+          case 1:
+            weight = UIImageSymbolWeightThin;
+            break;
+          case 2:
+            weight = UIImageSymbolWeightLight;
+            break;
+          // 3 is regular
+          case 4:
+            weight = UIImageSymbolWeightMedium;
+            break;
+          case 5:
+            weight = UIImageSymbolWeightSemibold;
+            break;
+          case 6:
+            weight = UIImageSymbolWeightBold;
+            break;
+          case 7:
+            weight = UIImageSymbolWeightHeavy;
+            break;
+          case 8:
+            weight = UIImageSymbolWeightBlack;
+            break;
+          default:
+            weight = UIImageSymbolWeightRegular;
+            break;
+        }
+
+        UIImageSymbolConfiguration *pointSizeConfig =
+            [UIImageSymbolConfiguration configurationWithPointSize:pointSize
+                                                            weight:weight
+                                                             scale:scale];
+
+        UIImage *finalImage;
+
+        if (@available(iOS 15, *)) {
+          NSNumber *preferMulticolor = call.arguments[5];
+          UIImageSymbolConfiguration *colors;
+
+          if ([preferMulticolor boolValue]) {
+            colors = [UIImageSymbolConfiguration configurationPreferringMulticolor];
+          } else {
+            colors = [UIImageSymbolConfiguration configurationWithPaletteColors:colorArray];
```

Comment:
What does this do if the array is empty?  And is the renderingMode set up correctly (maybe it is when you use  `-systemImageNamed:withConfiguration:` I've never used that API though)?

https://developer.apple.com/documentation/uikit/uiimagesymbolconfiguration/3810054-configurationwithpalettecolors?language=objc it says 

> For this color configuration to have an effect, your symbol image must have the following:
> - Its [renderingMode](https://developer.apple.com/documentation/uikit/uiimage/1624122-renderingmode?language=objc) set to [UIImageRenderingModeAlwaysTemplate](https://developer.apple.com/documentation/uikit/uiimagerenderingmode/uiimagerenderingmodealwaystemplate?language=objc) or [UIImageRenderingModeAutomatic](https://developer.apple.com/documentation/uikit/uiimagerenderingmode/uiimagerenderingmodeautomatic?language=objc).
> - Hierarchical layer annotations. If your symbol doesn’t have hierarchical layer annotations, the resulting image is a monochrome (template) symbol image.